include/zephyr/bluetooth/audio/has.h

```diff
@@ -95,6 +107,61 @@ struct bt_has_client_cb {
 	 * @param index Active preset index.
 	 */
 	void (*preset_switch)(struct bt_has *has, uint8_t index);
+
+	/**
+	 * @brief Callback function for presets read operation.
+	 *
+	 * The callback called when the preset record update is notified by the remote server.
+	 * The record object as well as its objects are temporary and must be copied to in order
+	 * to cache its information.
+	 *
+	 * @param has Pointer to the Hearing Access Service object.
+	 * @param err 0 on success, ATT error or negative errno otherwise.
+	 * @param record Preset record or NULL on errors.
+	 * @param is_last True if Read Presets operation can be considered concluded.
+	 */
+	void (*preset_read_rsp)(struct bt_has *has, int err,
+				const struct bt_has_preset_record *record, bool is_last);
+
+	/**
+	 * @brief Callback function for preset update notifications.
+	 *
+	 * The callback called when the preset record update is notified by the remote server.
+	 * The record object as well as its objects are temporary and must be copied to in order
+	 * to cache its information.
+	 *
+	 * @param has Pointer to the Hearing Access Service object.
+	 * @param index_prev Index of the previous preset in the list.
+	 * @param record Preset record.
+	 * @param is_last True if preset list update operation can be considered concluded.
+	 */
+	void (*preset_update)(struct bt_has *has, uint8_t index_prev,
+			      const struct bt_has_preset_record *record, bool is_last);
+
+	/**
+	 * @brief Callback function for preset deletion notifications.
+	 *
+	 * The callback called when the preset has been deleted by the remote server.
+	 *
+	 * @param has Pointer to the Hearing Access Service object.
+	 * @param index Preset index.
+	 * @param is_last True if preset list update operation can be considered concluded.
```

Comment:
Not really, the Server might send multiple Preset Changed notifications like:
`Generic Update` followed by `Preset Available` etc. and the last one has to have `is_last` flag set, so that the preset list update operation can be considered concluded

Reply:
Still not clear to me.
There are three callbacks, `preset_update`, `preset_deleted` and `preset_availability`.  

The `preset_deleted` is for "preset deletion notifications", but the `is_last` parameter refers to the "update operation".
Similar for the `preset_availability`.

Does the "update" operation have three possible callbacks, of which only one is called update?
And are preset deletion and preset availability part of the preset update?

Reply:
Then multiple of the preset records changed on the server the server sends the series of notifications that carry information about *updates in the list* of its internal records.

The notification is called *Preset Changed* notification and one of it's parameters is *ChangeID*. The *ChangeID* determines which callback will be called on our side. Once the last *Preset Changed* notification is received, appropriate callback is called (whether the last one *ChangeID* was Preset Deleted, Generic Update, Preset Available or Preset Unavailable) with the `is_last` flag set.

If that is still confusing I can make it as a single callback (it was done like this initially, see comment https://github.com/zephyrproject-rtos/zephyr/pull/45071#discussion_r858654067 but we agreed to split it to multiple callbacks)